test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs

```
@@ -0,0 +1,328 @@
+                {
+                    await connection.StartAsync().OrTimeout();
+
+                    // Stop, but wait at the sync point.
+                    var disposeTask = connection.StopAsync().OrTimeout();
+                    await syncPoint.WaitForSyncPoint();
+
+                    // Now start invoking the method under test
+                    var targetTask = method(connection).OrTimeout();
+                    Assert.False(targetTask.IsCompleted);
+
+                    // Release the sync point
+                    syncPoint.Continue();
+
+                    // Wait for the method to complete, with an expected error.
+                    var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => targetTask);
+                    Assert.Equal($"The '{name}' method cannot be called if the connection is not active", ex.Message);
+
+                    await disposeTask;
+                });
+            }
+
+            private static async Task ForceLastInvocationToComplete(TestConnection testConnection)
+            {
+                // We need to "complete" the invocation
+                var message = await testConnection.ReadSentTextMessageAsync();
+                var json = JObject.Parse(message); // Gotta remove the record separator.
+                await testConnection.ReceiveJsonMessage(new
+                {
+                    type = HubProtocolConstants.CompletionMessageType,
+                    invocationId = json["invocationId"],
+                });
+            }
+
+            // A helper that we wouldn't want to use in product code, but is fine for testing until IAsyncDisposable arrives :)
+            private static async Task AsyncUsing(HubConnection connection, Func<HubConnection, Task> action)
+            {
+                try
+                {
+                    await action(connection);
+                }
+                finally
+                {
+                    await connection.DisposeAsync();
```

OrTimeout?

---

So, I had `.OrTimeout` here before, but the idea here was that tests that use this aren't actually testing `DisposeAsync`. A timeout error in the finally block would suppress any exception from the test itself, so I was actually thinking of firing-and-forgetting this.

---

Added a comment to that effect. Happy to continue the discussion on whether we should `.OrTimeout` here though.